### DISCUSSION TITLE
Suggestions

### DISCUSSION THREAD

**Discussion by TEOL85** [intel/openvino-plugins-ai-audacity#154] [Ideas]
Well 1st Noise Suppression needs to get a bit smarter because I used it on a tape recording thats mainly hiss and it didn't take away the hiss it took away the music. Maybe we can in the next version copy a hiss to the clipboard and the noise suppression would only remove that so it's working exactly the same as the bog standard but at AI Levels of smartness and it also needs to drop Suppression in the title because that word has a tainted reputation in the electronics world it was first a really muddy sounding alternative to Dolby Chips on the later Tape players and then it became a feature in Windows which doesn't work and made your voice recordings very watery and robotic as well as Mono sounding that everyone turns it off and excepts the hiss without it.

2nd I would actually like some smart audio remastering where it picks up what the track requires and adds it automatically I think that would turn Audacity into something professional for free and if a track is below remastering in one go you can keep running it until it does sound decent without distortion as in Audacity you can reverse processes if it's distorted anyway of course having 24LUFS, 23LUFS and 16LUFS all as volume mastering on every boot up is a must because then a professional will never do Loudness Wars again but have one of the 3 big universal standards on their smart mastering to begin with.

3rd maybe a professional feature for Whisper out of all the apps where your labels can become cloned voices in your language making it the ideal AI application for the future for UK television companies on the international front BBC & Channel 4 being able to put their European shows with the same accents talking in English in a dubbed version and what about Utsav, Colors and Zee on the Satellite and Cable end being able to provide us with full dubs for the first time where turning on the subs is only for the hard of hearing or understanding of the Indian accent but if you understand the parts where they speak in English anyway you'll have no issue understanding anyway. And the amount of ANIME they can bring to the UK in original form factor and voices as well and Japanese wrestling can probably get a channel as well that would be a game changing professional feature all for free.

4th of course we gotta have the technology used in the 3rd one in something else the AI Music Producer. We first gotta get that doing the Sonauto smart prompt creating from minimal prompts where after your first generation your prompt opens up more in depth especially if you have a backing track for inspirations where it detects what the instruments are and adapts the prompt like Sonauto does. Then I feel it needs maybe a mixture of Suno AI's and Udio's instrumentation smartness. Definitely Udio style vocal creation and vocal performance where it doesn't sound robotic because here is the feature idea guys but it needs to get on the level of the big ones on the music first for this to stand a chance and the vocal needs to be Udio quality for this feature to work ok now I'll drop the feature but then number 3 was a clue as well did you guess it correct? Yep your talking voice to good singing if you can't sing you can sing now but when doing it live you would have to learn mime though voice cloning for your own music.

5th of course bringing lyrics into music generation to start and in the Google Deepmind quality of Udio in it's very first build asking who would like to be involved in the project to make the best audio generation app actually out there actually available on your audio editor.

Of course with the good ones often failing when you click generate intro or outro at times where it doesn't give you a stop or fade out ending but gives you more track you don't want because you have reached the 3 minutes 45 point and it's 4 minutes 22 max it is annoying audacity Open Vinu needs to have smarter endings as it's in an auto editor though we can have unlimited length if we like. The intros thing normally does the generation perfect though in the Udio but if you feel you want more words before the current intro of the vocal bits after that it doesn't smartly splices it in but puts it before the intro no you have ruined your track. lol.

Yeah we need to get competitive with the rivals on music generation especially if that new Udio ends up being the free limits now for always free and their paid options double and tripling the generations and making the lengths 12 inchers and Udio being good for pretty much everything on the free Beta other than auto prompt adapting on first generation to only a few words that goes to Sonauto which is totally free but only 90 seconds. But in Udio you have a full decision on if you want to go with your own lyrics set for a section or an instrumental section for a section or auto generated lyrics so it's really smart however I have had outros where I choose Instrumental in my freedoms and got vocals and a track continuation not an ending.

I do think the music generation is under par to what it is for free on the rivals it's only basic and those into AI Music doesn't really want basic they want semi pro and professionals want completely pro

For free in Suno you get to adapt your production for about 2 minutes a day until you reached the max length limit anyway and it's full professional quality backgrounds but the voice is a bit robot.

For free in Udio Beta you get proper voice, freedom on instrumental, custom and auto generated lyrics as well as where you want to place that new part in your track and real professional intros and outros.

In Sonauto you got put your own track into the generated track features with a slider more to the inspiration or more to the prompt and the prompt if you start with Glam Rock and 1970s actually generates the prompt longer and mentions how your background music is and puts it all into your prompt but only generates 90 seconds.

In Audacity if we get all the features that the others get within free limits. Open Vinu will be the professionals choice especially if it uses it's own voice cloning and puts your voice as the vocalist over the one generated in the generation process and unlimited length as Audacity is free and the endings actually stop it proper.

Lastly I do hope we can get a proper installer the current one takes you to a slow website downloading the models at a snails pace we just want to get into installing the stuff we choose and it flying through not taking hours getting a model.

I forgot one more suggestion the AI Stems Spitter again I think that needs to be smarter as well 4 stems is not good enough I want an auto mode where it detects different instruments and different voices and stems them into 8 layers into 16, into 32, into 64.

4 Channels is Commodore AMIGA 1985 we are in the days where the different instruments are in way more channels than a humble AMIGA so we need more smarter splitting just an auto split if it's 64 melodies of instruments thats what it gives us instead of this other instruments bit.

What if it's a track we really want to take something out like Beetles did on Now And Then if there is a sound effect we no longer want in the track and want to rip out we can do with smarter splitting if it's put into a split with aspects you don't want took out of the track because it's put into other instead of being split into it's own SFX section.

And then if you really want to delve deeper vocals on top of vocals a bit of talking on top of the track with the singing in it and both of them end up on top of each other in the processed.

**Comment by MarcoRavich** [comment]
Bump.

About noise suppression: I don't know which model has been used actually, but I believe that users should have the ability to choose between speech and audio denoising (which are different beasts).

Here are some related resources I've collected for the [HyMPS project](https://forart.it/HyMPS):
- [AUDIO](https://github.com/FORARTfe/HyMPS#-) \ [AI-based](https://github.com/FORARTfe/HyMPS/blob/main/Audio/AI-based.md#--) \ [Enhancers](https://github.com/FORARTfe/HyMPS/blob/main/Audio/AI-Enhancing.md#---) \ [Denoisers](https://github.com/FORARTfe/HyMPS/blob/main/Audio/AI-Enhancing.md#denoisers-);
- [AUDIO](https://github.com/FORARTfe/HyMPS#-) \ [AI-based](https://github.com/FORARTfe/HyMPS/blob/main/Audio/AI-based.md#--) \ [Voicing](https://github.com/FORARTfe/HyMPS/blob/main/Audio/AI-Voicing.md#---) \ [Denoisers](https://github.com/FORARTfe/HyMPS/blob/main/Audio/AI-Voicing.md#denoisers-).

I honestly don't know which is better but, for example, [NVIDIA choosed to implement CleanUNet model](https://github.com/NVIDIA/CleanUNet#readme) for speech denoising.

**Comment by MarcoRavich** [comment]
Bump 2.

About mastering: except @sergree's [Matchering](https://github.com/sergree/matchering#readme) (that seems more a _style transfer_ tool), there are really few open projects about.
Anyway @SUC-DriverOld has recently realeased [Open Soren](https://github.com/SUC-DriverOld/Open-Soren#readme) - along with [some models](https://github.com/SUC-DriverOld/Open-Soren/tree/master/model) - but I haven't tested it.

As always, any project I found is listed under the [AUDIO section](https://github.com/FORARTfe/HyMPS#-) \ [AI-based page](https://github.com/FORARTfe/HyMPS/blob/main/Audio/AI-based.md#--) for the [HyMPS project](https://forart.it/HyMPS)'s resource collection.

Hope that inspires !

**Reply by MarcoRavich** [reply]
Open Soren (which was a code leak) gone...

**Comment by TEOL85** [comment]
I think OpenVinu might need to go a different approach for a full Dolby Nr hiss and hum ripoff for tapes and a Vinyl Crackle and deck Rumble and Hiss removal.

Using tape and vinyl recordings with no treatment on them making a load of noise profiles and then using those noise profiles to get someone who knows what they are doing to write an AI model based off of all the profiles might be the correct way to do this rather than straight up borrowing existing open source models which might not do it so perfect yet.

As for AI Music models Diffusion Open Source has gotten really good now now actaully matching Suno and Udio being able to make a proper tune for you we need to update the outdated model.

As for the Super Resolution works crap for me on i7 hardware with 8GB RAM I have the fan spinning non stop 30 minutes just for 10 seconds of work just for it to casuse a crash on that 10 seconds as well to make it usable it has to work on this kind of hardware a full track in 5 minutes max and not cause any crashes on your hardware at one point the monitor crashed out on processing and the mouse wouldn't move it's unusable.

**Reply by RyanMetcalfeInt8** [reply]
> Using tape and vinyl recordings with no treatment on them making a load of noise profiles and then using those noise profiles to get someone who knows what they are doing to write an AI model based off of all the profiles might be the correct way to do this rather than straight up borrowing existing open source models which might not do it so perfect yet.

If 'someone' trains a model for this, let us know.

> As for AI Music models Diffusion Open Source has gotten really good now now actaully matching Suno and Udio being able to make a proper tune for you we need to update the outdated model.

Which specific open source models are you referring to here that 'match' Suno and Udio?

> As for the Super Resolution works crap for me on i7 hardware with 8GB RAM I have the fan spinning non stop 30 minutes just for 10 seconds of work just for it to casuse a crash on that 10 seconds as well to make it usable it has to work on this kind of hardware a full track in 5 minutes max and not cause any crashes on your hardware at one point the monitor crashed out on processing and the mouse wouldn't move it's unusable.

It's hard to say what's could be going wrong. I suspect that the Super Resolution models will consume too much of a memory footprint to reasonably fit on your 8GB RAM system. Also, there are many generations of 'i7 hardware', it'd be helpful to understand which one you're using.

**Reply by MarcoRavich** [reply]
> I think OpenVinu might need to go a different approach for a full Dolby Nr hiss and hum ripoff for tapes and a Vinyl Crackle and deck Rumble and Hiss removal.
> 
> Using tape and vinyl recordings with no treatment on them making a load of noise profiles and then using those noise profiles to get someone who knows what they are doing to write an AI model based off of all the profiles might be the correct way to do this rather than straight up borrowing existing open source models which might not do it so perfect yet.
> 

Keep in mind that these OpenVINO audio plugs "just" perform inference using pretrained models, but does not allows users to train them.

> As for AI Music models Diffusion Open Source has gotten really good now now actaully matching Suno and Udio being able to make a proper tune for you we need to update the outdated model.
> 

Both are pay-per-use (online) services and they don't provide any model, so it's not a way to go.

> As for the Super Resolution works crap for me on i7 hardware with 8GB RAM I have the fan spinning non stop 30 minutes just for 10 seconds of work just for it to casuse a crash on that 10 seconds as well to make it usable it has to work on this kind of hardware a full track in 5 minutes max and not cause any crashes on your hardware at one point the monitor crashed out on processing and the mouse wouldn't move it's unusable.
>

...WHICH i7 ?

Even if I haven't tried SR yet, others - such as steam separation - works pretty well on my i5-12500T/32Gb: in both terms of results quality and performances it pairs Izotope's RX 11, but for free.

**Comment by timeship12** [comment]
**"WHO OWNS OUR CLONED VOICES?"**

Can't wait for Audacity plugins to clone our own voice for audiobook narrators. After all, our cloned voices are our personal intellectual property, and no 2nd-party online services have the right to ask us to PAY to use our own voices. This is simply a legalized intellectual crime and subscription slavery. Our voices should be copyright-owned by us. If anyone knows how to do that, tell me how to free the overly exploited "cow," which is us! THX

**Reply by MarcoRavich** [reply]
Well, to be honest, many (if not all) online services rents you computational power to inference...

**Comment by TEOL85** [comment]
We may be able to use OpenVinu in any new audacity from now on but surely we should still be getting new versions with new tools or upgrades to the existing ones guys it seems like now it's made to work with all versions the devs have become stagnant in fact now I think there is a Beta for a totally new version 4.0 and the current 3.0 version parallel for current versions.

**Reply by RyanMetcalfeInt8** [reply]
New updates are coming for Audacity 3.X version, it's just taking me a while to finalize it. We are actively developing new features on `next` branch here: https://github.com/intel/openvino-plugins-ai-audacity/tree/next . Folks can always try it out if they are willing to build from source.

**Comment by TEOL85** [comment]
You gave me hope today for a new update to allow for getting a sample to ai lossless noise remove from a bit of music as well. I hope it's available for mixed platforms 3 & 4 I hope for the stem splitter to go more suno as well as 4 channels isn't really enough although good for less damage noise removal on the rubbish one because the ai one can't do it on music yet no get sample button to get a noise on a profile which the ai removal would nuke without getting rid of the rest.. lol
